LSPS5/README.md

```diff
@@ -0,0 +1,264 @@
+
+```JSON
+{
+   "versions": [1]
+}
+```
+
+`versions` is an array of versions of this API that the LSP
+supports.
+
+### Webhook Registration API
+
+The client can register a specific webhook for the LSP to
+contact in case the client is asleep, but the LSP needs the
+client to be online and ready to interact.
+
+The client uses the `lsps5.setwebhook` call to specify the
+HTTPS URI that the LSP should `POST` in order to send a
+push notification to the client user.
+It takes the parameters below:
+
+```JSON
+{
+    "version": 1,
+    "webhook": "https://www.example.org/push?l=1234567890abcdefghijklmnopqrstuv&c=42"
+}
+```
+
+`version` is a required integral number, the version of the
+API that the client wants to use.
+
+`webhook` is a required string, containing the URI of the
+webhook that the LSP can use to push a notification to the
+client.
+
+The client:
+
+* MUST use a HTTPS-protocol URI to the client application
+  server.
+* MUST encode any data needed to authorize a particular
+  LSP to contact this webhook as a `GET` parameter in the
+  URI (i.e. the parameters after `?` in the URI).
+  The later `PUSH` notification will include the LSP
+  public node ID and a signature from the LSP.
+  * For example, if the client developer wants to sign
+    off on LSPs who are allowed to push to users of the
+    client, the client can encode that signature as a `GET`
+    parameter in the given URI.
+* MUST encode any data neded to identify the mobile
+  device as a `GET` parameter.
+
+> **Rationale** By placing any data to the `GET` parameters
+> of the given URI, the LSP interface is simplified and
+> the client developer has the flexibility to add any
+> additional information they want the LSP to simply
+> pass through from the client instance to the client
+> application server.
+
+The LSP:
+
+* MUST persist the given webhook and associate it with the
+  client node ID before responding to this call.
+* MUST use the `webhook` specified in the latest
+  `lsps5.setwebhook` call.
+* MAY forget the `webhook` after 7 days (7 x 24 hours) from
+  the definite closure of the last channel between the LSP
+  and the client.
+  * MUST treat a possible channel open, such as via a
+    reserved SCID for a JIT channel, as a channel open, for
+    purposes of remembering the `webhook` for a client.
+
+The `lsps5.setwebhook` call results in the empty object
+`{}` on success, and has the following errors defined:
+
+* `unsupported_version` (1) - The `version` specified by
+  the client is not among the `versions` that the LSP
+  supports.
+* `unsupported_protocol` (2) - The `webhook` is not an
+  HTTPS-protocol URI.
+
+### Webhook Call
+
+The LSP SHOULD make a `POST` request to the given webhook
+URI, if the client has registered one via `lsps5.setwebhook`
+*and* one of the following conditions is true:
+
+* The LSP learns of an incoming payment to the client
+  (whether an incoming HTLC, or other future mechanism
+  for incoming payments).
```

Comment:
Not really want to nail down timeouts.  HTLCs should be shorter (like a minute or two at most) but something like an asynchronous receive could usefully be held for a day before failing (assuming async receive has a concept of the receiver LSP failing it, that is).

Since HTLCs are supported today, maybe just nail down the timeout for HTLC and leave future possible timeouts to future versions:

```Markdown
* The LSP learns of an incoming payment to the client
  (whether an incoming HTLC, or other future mechanism
  for incoming payments).
  * If the LSP learns via an incoming HTLC for the client,
    it SHOULD wait for the client to be able to complete
    the receive for up to 60 to 120 seconds.
    Payment completion may be by normal HTLC forwarding,
    LSPS2 JIT Channels, or any future LSPS-defined method
    of forwarding payments.
```

Reply:
Yeah I can see how you might not want to burn yourself on timeouts here. Maybe just a general indication: it has to be long enough to allow the client to come online, but short enough to not lock up liquidity in the network or allow htlcs to expire.

Reply:
Went with a vaguer text about long enough to let the client come online, but short enough to not lock up liquidity.